From 741243f707cac7de1a3769a38e03004f037f4a3d Mon Sep 17 00:00:00 2001
From: iglocska <andras.iklody@gmail.com>
Date: Tue, 19 Jan 2021 14:09:17 +0100
Subject: [PATCH] fix: [security] Stored XSS in the galaxy cluster view

- Galaxy cluster names were vulnerable to XSS injection

- As reported by Patrik Kontura of ESET
---
 app/View/GalaxyClusters/view.ctp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/app/View/GalaxyClusters/view.ctp b/app/View/GalaxyClusters/view.ctp
index 1ef8e02aef..ec53a34f7f 100755
--- a/app/View/GalaxyClusters/view.ctp
+++ b/app/View/GalaxyClusters/view.ctp
@@ -58,15 +58,15 @@
         'field' => array('data_path' => 'distribution')
     ));
     $table_data[] = array(
-        'key' => __('Owner Organisation'), 
+        'key' => __('Owner Organisation'),
         'html' => $this->OrgImg->getOrgImg(array('name' => $cluster['GalaxyCluster']['Org']['name'], 'id' => $cluster['GalaxyCluster']['Org']['id'], 'size' => 18), true),
     );
     $table_data[] = array(
-        'key' => __('Creator Organisation'), 
+        'key' => __('Creator Organisation'),
         'html' => $this->OrgImg->getOrgImg(array('name' => $cluster['GalaxyCluster']['Orgc']['name'], 'id' => $cluster['GalaxyCluster']['Orgc']['id'], 'size' => 18), true),
     );
     $table_data[] = array('key' => __('Connector tag'), 'value' => $cluster['GalaxyCluster']['tag_name']);
-    $table_data[] = array('key' => __('Events'), 'html' => isset($cluster['GalaxyCluster']['tag_count']) ? 
+    $table_data[] = array('key' => __('Events'), 'html' => isset($cluster['GalaxyCluster']['tag_count']) ?
                         sprintf('<a href="%s">%s</a>',
                             sprintf('%s/events/index/searchtag:%s', $baseurl, h($cluster['GalaxyCluster']['tag_id'])),
                             __n('%s event', '%s events', $cluster['GalaxyCluster']['tag_count'], h($cluster['GalaxyCluster']['tag_count']))
@@ -85,7 +85,7 @@
     <div class="row-fluid">
         <div class="span8">
             <h2>
-                <?= sprintf('%s :: %s', h($cluster['GalaxyCluster']['Galaxy']['name']), $cluster['GalaxyCluster']['value']); ?>
+                <?= sprintf('%s :: %s', h($cluster['GalaxyCluster']['Galaxy']['name']), h($cluster['GalaxyCluster']['value'])); ?>
             </h2>
             <?php echo $this->element('genericElements/viewMetaTable', array('table_data' => $table_data)); ?>
         </div>
